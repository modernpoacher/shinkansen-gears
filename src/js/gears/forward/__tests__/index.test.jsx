import React from 'react'
import PropTypes from 'prop-types'
import snapshotOf, {
  getComponentElement
} from 'react-component-snapshot'

import '@testing-library/jest-dom'

import {
  render
} from '@testing-library/react'

import Forward from '#gears/gears/forward'

/**
 *  @param {{ to: string | { pathname: string }, children: React.ReactNode | React.ReactNode[] }} props
 *  @returns {React.JSX.Element}
 */
function MockLink ({ to, children }) {
  if (typeof to === 'string') {
    return (
      <a href={to} className='mock-link'>
        {children}
      </a>
    )
  }

  const {
    pathname
  } = to

  return (
    <a href={pathname} className='mock-link'>
      {children}
    </a>
  )
}

MockLink.propTypes = {
  to: PropTypes.oneOfType([
    PropTypes.string,
    PropTypes.shape({})
  ]),
  children: PropTypes.oneOfType([
    PropTypes.node,
    PropTypes.arrayOf(
      PropTypes.node
    )
  ])
}

jest.mock('react-router', () => {
  return {
    __esModule: true,
    Link: MockLink
  }
})

describe('#gears/gears/forward', () => {
  it('renders', () => {
    expect(snapshotOf(getComponentElement(render(
      <Forward
        pathname='MOCK PATHNAME'
      />
    ))))
      .toMatchSnapshot()
  })
})
