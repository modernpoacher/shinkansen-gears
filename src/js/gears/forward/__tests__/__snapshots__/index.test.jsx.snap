// Jest Snapshot v1, https://jestjs.io/docs/snapshot-testing

exports[`#gears/gears/forward renders 1`] = `
<li
  className="shinkansen-gears-forward"
>
  <MockLink>
    <a
      className="mock-link"
      href="MOCK PATHNAME"
    >
      Forward
    </a>
  </MockLink>
</li>
`;
